wpengine/faustjs
# Implement experimental plugin system

## Description

This PR introduces the plugin system using [@wordpress/hooks](https://www.npmjs.com/package/@wordpress/hooks) behind an `experimental` flag.

## Testing
1. Checkout the PR
2. Run `npm run build -w faust-nx`
3. Replace the `examples/next/faust-nx/faustnx.config.js` file contents with:
   ```js
    import { setConfig } from 'faust-nx';
    import templates from './wp-templates';
    
    class HelloWorld {
      apply(hooks) {
        console.log(hooks);
      }
    }
    
    /**
     * @type {import('faust-nx').FaustNXConfig}
     **/
    export default setConfig({
      templates,
      experimentalPlugins: [new HelloWorld()],
    });
   ```
4. Run `npm run dev -w faust-nx-getting-started`
5. Notice the plugin's `apply()` method gets called and the `hooks` argument gets consoled.
